bokeh/client/session.py

```diff
@@ -7,11 +7,36 @@
 log = logging.getLogger(__name__)
 
 from ..document import Document
-from ..resources import DEFAULT_SERVER_WEBSOCKET_URL, server_url_for_websocket_url, _SessionCoordinates
+from ..resources import _SessionCoordinates, DEFAULT_SERVER_HTTP_URL
 from ..util.session_id import generate_session_id
 
 DEFAULT_SESSION_ID = "default"
 
+def websocket_url_for_server_url(url):
+    if url.startswith("http:"):
+        reprotocoled = "ws" + url[4:]
+    elif url.startswith("https:"):
+        reprotocoled = "wss" + url[5:]
+    else:
+        raise ValueError("URL has unknown protocol " + url)
+    if reprotocoled.endswith("/"):
+        return reprotocoled + "ws"
+    else:
+        return reprotocoled + "/ws"
+
+def server_url_for_websocket_url(url):
+    if url.startswith("ws:"):
+        reprotocoled = "http" + url[2:]
+    elif url.startswith("wss:"):
+        reprotocoled = "https" + url[3:]
+    else:
+        raise ValueError("URL has non-websocket protocol " + url)
+    if not reprotocoled.endswith("/ws"):
+        raise ValueError("websocket URL does not end in /ws")
+    return reprotocoled[:-2]
+
+DEFAULT_SERVER_WEBSOCKET_URL = websocket_url_for_server_url(DEFAULT_SERVER_HTTP_URL)
+
```

Comment:
Python client connections do need to construct full websocket URLs, AFAIK? But in that case move that code here under `bokeh.client`.